src/test/java/teammates/test/pageobjects/AppPage.java

```diff
@@ -818,7 +818,7 @@ boolean selectDropdownByVisibleValue(WebElement element, String text) {
      * @see AppPage#selectDropdownByVisibleValue(WebElement, String)
      */
     void selectDropdownByVisibleValueAndWaitForAjaxRequestComplete(WebElement element, String text) {
```

Comment:
Maybe we should change this to something like:
```Java
    /**
     * Waits for all ongoing AJAX requests to complete if any before selecting the option by visible text, then
     * waits for the associated AJAX request to complete.
     *
     * @see AppPage#selectDropdownByVisibleValue(WebElement, String)
     */
    void selectDropdownByVisibleValueAndHandleAjaxRequests(WebElement element, String text) {
```

I guess now the method is not really understandable anymore and making it any longer is probably a bad idea. But I suppose the new frontend rewrite should not be affected by this problem anymore (and if there are any similar cases we can probably use ideas like builders with mandatory constructors or step builders).